src/custom-element.js

```diff
@@ -797,6 +800,7 @@ export function createAmpElementProto(win, name, opt_implementationClass) {
         this.implementation_.firstLayoutCompleted();
       }
     }, reason => {
+      this.loadingError_ = true;
       this.toggleLoading_(false, /* cleanup */ true);
```

Comment:
`this.toggleLoading_(true)` in `viewportCallback` could be called later than this